lib/ecto/type.ex

```diff
@@ -602,7 +602,7 @@ defmodule Ecto.Type do
     {:ok, nil}
   end
 
-  def load({:maybe, type}, value, loader) do
+  def load({:aggregate, type}, value, loader) do
```

Comment:
There is one thing I don’t understand. If aggregate float falls back to float, why is it not loading as expected?

Reply:
Oh so what I found is the following happens for aggregates ([reference](https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/myxql.ex#L152C2-L157C35))

1. `Ecto.Adapters.MyXQL.loaders/2` is called with the following arguments `{:maybe, :float}` and `{:maybe, :float}`. That causes the return value to be `[{:maybe, :float}]`
2. The result from above eventually gets passed to [Ecto](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/type.ex#L605). Then Ecto tries to convert the decimal to float but it's not allowed. So it returns the decimal. 

Now with this change instead of (1) returning `[{:maybe, :float}]` it returns `[&float_decode/1, {:maybe, :float}]`. So first `float_decode/1` converts the decimal into a float. And then the `{:maybe, :float}` part will essentially be the identity function. 

I'm guessing the decimal -> float conversion is not allowed in Ecto itself because for most adapters it will be an unsafe downcast. It seems like MySQL allows it because [the precision is limited in that database](https://dev.mysql.com/doc/refman/8.0/en/precision-math-decimal-characteristics.html) which I'm guessing fits nicely into Elixir floats.

Reply:
I am surprised we are not falling back to calling loaders(:float) in the adapter. Perhaps that’s the change we need to make.

Reply:
I think if we add `def type({:aggregate, sub}), do: sub`, then it will work. as it will force us to pass the real type to adapters. But I am not sure what else we break. if that's the case, then perhaps I would even keep it as `:maybe`. Otherwise this PR is the best way to go indeed.

Reply:
> I think if we add def type({:aggregate, sub}), do: sub, then it will work. as it will force us to pass the real type to adapters. But I am not sure what else we break.

Going this route makes things a little confusing to me at least. But let me know what you think. When we get to this part:

```elixir
process_loaders(adapter.loaders(type(type), type), {:ok, value}, adapter)
```

We want `adapters.loaders` to be unaware of the `maybe`. But then we want `process_loaders` to be aware of it. So there is a conflict between the return value of `adapter.loaders` and what we want to happen after.

Reply:
Actually sorry I see what you are saying. Maybe this works I will play around with it.

Reply:
It seems like it's not breaking anything. Pushing the change to let the integration tests run.

Reply:
The integration tests pass. When looking through the code I can't see any other place that would break even if tests are missing. `{:maybe, type}` is only set in the planner during `collect_fields` and it's private otherwise. So besides loading/dumping aggregations from the database I don't believe there is any other hidden landmine.

It looks to me like you found the best change.

Reply:
Apologies for not seeing it sooner. :(

Reply:
I only realized it when review the PR :)

Reply:
No worries! Sorry I didn't see it either.